subprojects/launcher/src/test/groovy/org/gradle/tooling/internal/provider/FileSystemWatchingBuildActionRunnerTest.groovy

```diff
@@ -86,4 +96,39 @@ class FileSystemWatchingBuildActionRunnerTest extends Specification {
         WatchMode.DISABLED | VfsLogging.NORMAL  | WatchLogging.NORMAL | false
         WatchMode.DISABLED | VfsLogging.NORMAL  | WatchLogging.DEBUG  | false
     }
+
+    def "watching enabled by default is disabled when project cache dir is specified"() {
+        _ * startParameter.watchFileSystemMode >> WatchMode.DEFAULT
+        _ * startParameter.projectCacheDir >> Mock(File)
```

Comment:
What's your argument against mocking a `File` specifically?

I considered the other options, and I think I still prefer the `Mock`. Here's my line of reasoning:

- I want a fake object, because this unit test should not create files on disk,
- Using a fake object captures the fact that we don't want the production code to rely on the properties of this object, just on its identity.
- More specifically in this case, we don't want the code to even query the path of the file in question, and I think that expectation is better captured via a `Mock`, where the production code doing such a query would make the test fail.